packages/flutter/test/material/dialog_theme_test.dart

```diff
@@ -548,4 +550,54 @@ void main() {
       ),
     );
   });
+
+  testWidgets('DialogTheme.clipBehavior updates the dialogs clip behavior', (WidgetTester tester) async {
+    const DialogTheme dialogTheme = DialogTheme(
+      clipBehavior: Clip.hardEdge
+    );
+    const Dialog dialog = Dialog(child: Placeholder());
+
+    await tester.pumpWidget(_appWithDialog(
+      tester,
+      dialog,
+      theme: ThemeData(dialogTheme: dialogTheme),
+    ));
+
+    final Material materialWidget = _getMaterialFromDialog(tester);
+    expect(materialWidget.clipBehavior, Clip.hardEdge);
+  });
+
+  testWidgets('Material3 - Dialog.clipBehavior takes priority over theme', (WidgetTester tester) async {
+    const Dialog dialog = Dialog(
+      clipBehavior: Clip.antiAlias,
+      child: Placeholder()
+    );
+    final ThemeData theme = ThemeData(
+      useMaterial3: true,
+      dialogTheme: const DialogTheme(clipBehavior: Clip.hardEdge),
+    );
+
+    await tester.pumpWidget(
+      _appWithDialog(tester, dialog, theme: theme),
+    );
+    final Material materialWidget = _getMaterialFromDialog(tester);
+    expect(materialWidget.clipBehavior, Clip.antiAlias);
+  });
+
+  testWidgets('Material2 - Dialog.clipBehavior takes priority over theme', (WidgetTester tester) async {
+    const Dialog dialog = Dialog(
+      clipBehavior: Clip.antiAlias,
+      child: Placeholder()
+    );
+    final ThemeData theme = ThemeData(
+      useMaterial3: false,
+      dialogTheme: const DialogTheme(clipBehavior: Clip.hardEdge),
+    );
+
+    await tester.pumpWidget(
+      _appWithDialog(tester, dialog, theme: theme),
+    );
```

Comment:
```suggestion
    await tester.pumpWidget(
      _appWithDialog(tester, dialog, theme: theme),
    );
    await tester.tap(find.text('X'));
    await tester.pumpAndSettle();

```

Tap to dialog and add a line